From 192ad7782f930d0e33a6b92702706027401b5038 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Sun, 29 Nov 2020 12:42:47 +0000
Subject: [PATCH] GCC: fix conflicting type for kDefaultHangWatchTime

In hang_watcher.h kDefaultHangWatchTime has type base::TimeDelta,
but hang_watcher.cc has auto. GCC fails with this:

../../base/threading/hang_watcher.cc:44:16: error: conflicting
declaration 'constexpr const auto
base::HangWatchScopeEnabled::kDefaultHangWatchTime'
   44 | constexpr auto HangWatchScopeEnabled::kDefaultHangWatchTime =
         |                ^~~~~~~~~~~~~~~~~~~~~
	 I
---
 base/threading/hang_watcher.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/base/threading/hang_watcher.cc b/base/threading/hang_watcher.cc
index 9ee8d1f..63fd44d 100644
--- a/base/threading/hang_watcher.cc
+++ b/base/threading/hang_watcher.cc
@@ -41,7 +41,7 @@ constexpr base::FeatureParam<int> kUIThreadLogLevel{
 constexpr base::FeatureParam<int> kThreadPoolLogLevel{
     &HangWatcher::kEnableHangWatcher, "threadpool_log_level", 0};
 
-constexpr auto HangWatchScopeEnabled::kDefaultHangWatchTime =
+constexpr base::TimeDelta HangWatchScopeEnabled::kDefaultHangWatchTime =
     base::TimeDelta::FromSeconds(10);
 
 namespace {
-- 
2.26.2

